code generators/de.fhdo.lemma.model_processing.code_generation.container_base/src/main/java/de/fhdo/lemma/model_processing/code_generation/container_base/code_generation/DockerFileCodeGenerator.xtend

```diff
@@ -0,0 +1,157 @@
+        // Receive the intermediate operation model
+        model = this.intermediateModelResource.contents.get(0) as IntermediateOperationModel
+
+        // Create a Map which contains the generated source code
+        content = new HashMap
+
+        // Create a dockerfile for containers
+        model.containers.forEach[container |
+            this.createDockerFileForContainer(container)
+        ]
+
+        // Create dockerfile for infrastructure nodes
+        model.infrastructureNodes.forEach[node |
+            val property = node.defaultValues.findFirst[property |
+                property.technologySpecificProperty.name == "imageType"
+            ]
+
+            if (property.value.toLowerCase != "dockerhub")
+                this.createDockerFileForInfrastructureNode(node)
+        ]
+
+        println("CodeGenerationModule dockerfile finished.")
+        return withCharset(content, "UTF-8");
+    }
+
+    /*
+     * Receive the language description for the intermediate operation model package.
+     */
+    @NotNull
+    override getLanguageDescription() {
+        new LanguageDescription(IntermediatePackage.eINSTANCE)
+    }
+
+    /*
+     * This method is responsible for the creation of the dockerfile.
+     * The creation of the dockerfile is only supported for container deploying a single service,
+     * because of an architectural point of view the deployment of more than one microservice in a
+     * container breaks the principle of an independent deployment
+     */
+    private def createDockerFileForContainer(IntermediateContainer container) {
+        if (container.deployedServices.size > 1) {
+            throw new DeployedServiceNumberExcpetion("The Number of deployed services in a single"
+                + "container exceeds the maximum of one service per container ")
+        }
+
+        /*
+         * The creation of the dockerfile is supported in two different ways.
+         * First we check is the dockerfile is explicitly defined in an aspect.
+         * Otherwise the dockerfile is build based on the name of the deployed microservice,
+         * operation environment and exposed port, defined by the container endpoints.
+         */
+        val dockerFileAspect = container.aspects.findFirst[aspect |
+            aspect.name.toLowerCase == "dockerfile"
+        ]
+
+        if (dockerFileAspect !== null) {
+            val dockerFile = new DockerFile(dockerFileAspect.propertyValues.get(0).value)
+
+            content.put(targetFolder
+                + File.separator
+                + Util.buildPathFromQualifiedName(container.deployedServices.get(0).qualifiedName)
+                + File.separator
+                + "Dockerfile",
+                dockerFile.toString
+            )
+        } else {
+            var dockerFile = new DockerFile
+            dockerFile.deployedServiceName = container.name.toLowerCase
+
+            if (container.deployedServices.get(0).version === null) {
```

Comment:
Please omit curly brackets whenever possible, like in this if-statement and its else-branch.